hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4526,35 +4526,32 @@ func (r *HostedClusterReconciler) reconcileDeprecatedGlobalConfig(ctx context.Co
 		return err
 	}
 
-	// Only copy over config from the raw extension if the field is not already populated.
-	// Once populated, the field takes precedence and a conflicting raw extension config
-	// results in an invalid configuration field.
-
-	if gconfig.APIServer != nil && hc.Spec.Configuration.APIServer == nil {
+	// Copy over config from the raw extension
+	if gconfig.APIServer != nil {
 		hc.Spec.Configuration.APIServer = &gconfig.APIServer.Spec
 	}
-	if gconfig.Authentication != nil && hc.Spec.Configuration.Authentication == nil {
+	if gconfig.Authentication != nil {
 		hc.Spec.Configuration.Authentication = &gconfig.Authentication.Spec
 	}
-	if gconfig.FeatureGate != nil && hc.Spec.Configuration.FeatureGate == nil {
+	if gconfig.FeatureGate != nil {
 		hc.Spec.Configuration.FeatureGate = &gconfig.FeatureGate.Spec
 	}
-	if gconfig.Image != nil && hc.Spec.Configuration.Image == nil {
+	if gconfig.Image != nil {
 		hc.Spec.Configuration.Image = &gconfig.Image.Spec
 	}
-	if gconfig.Ingress != nil && hc.Spec.Configuration.Ingress == nil {
+	if gconfig.Ingress != nil {
 		hc.Spec.Configuration.Ingress = &gconfig.Ingress.Spec
 	}
-	if gconfig.Network != nil && hc.Spec.Configuration.Network == nil {
+	if gconfig.Network != nil {
 		hc.Spec.Configuration.Network = &gconfig.Network.Spec
 	}
-	if gconfig.OAuth != nil && hc.Spec.Configuration.OAuth == nil {
+	if gconfig.OAuth != nil {
 		hc.Spec.Configuration.OAuth = &gconfig.OAuth.Spec
 	}
-	if gconfig.Scheduler != nil && hc.Spec.Configuration.Scheduler == nil {
+	if gconfig.Scheduler != nil {
 		hc.Spec.Configuration.Scheduler = &gconfig.Scheduler.Spec
 	}
-	if gconfig.Proxy != nil && hc.Spec.Configuration.Proxy == nil {
+	if gconfig.Proxy != nil {
 		hc.Spec.Configuration.Proxy = &gconfig.Proxy.Spec
 	}
 	hc.Spec.Configuration.Items = nil
```

Comment:
If the unstructured config will still take precedence, then we likely don't want to clear it here

Reply:
Gotcha ok. I'm fine with clearing the older fields, as long as the unstructured config gets get overlaid on top of the new fields on apply. But we can simply not clear too - just didn't know if we wanted to keep this mutating behavior.

Reply:
And also for my clarity, if we don't clear `items[]`, we would still want to populate the new fields?

Reply:
We definitely want to populate the new fields.
The reason I was thinking of not clearing the old fields is that if you have code updating the config there, then you're going to be fighting against the hc controller. One clearing it, the other populating it.

Reply:
> We definitely want to populate the new fields.

Understood!

> The reason I was thinking of not clearing the old fields is that if you have code updating the config there, then you're going to be fighting against the hc controller. One clearing it, the other populating it.

Ah good point, you maybe right there. Ok will update!

Reply:
One more thing! Do we want to keep the ConfigMapRefs and SecretRefs, as well? I see that for the HCP reconcilication, it now parses these info from the new fields https://github.com/openshift/hypershift/blob/704177c8855e34fd53d39d4c9b83696947c09edc/hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go#L1474-L1487

Reply:
I think it’s fine the way it is

Reply:
@csrwng Getting stuck on this one. When I add `hc.Spec.Configuration.Items = nil`, unit tests starts to err with..
```
# hypershift-operator/controllers/hostedcluster/hostedcluster_controller_test.go:2206

unexpected get error: json: error calling MarshalJSON for type runtime.RawExtension: invalid character 'a' looking for beginning of value
```
Do you happen to have any ideas?

Reply:
```
serializedAPIServer, err := util.SerializeResource(apiServer, hyperapi.Scheme)
```
This ^^ is serializing the apiserver as yaml, for a RawExtension it needs to be JSON
See https://github.com/openshift/hypershift/blob/main/hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go#L4765-L4781